src/QMCWaveFunctions/Fermion/DiracDeterminantBatched.cpp

```diff
@@ -664,8 +664,9 @@ typename DiracDeterminantBatched<DET_ENGINE>::GradType DiracDeterminantBatched<D
   {
     resizeScratchObjectsForIonDerivs();
     Phi->evaluateGradSource(P, FirstIndex, LastIndex, source, iat, grad_source_psiM);
-    // FIXME due padding of psiMinv
-    g = simd::dot(psiMinv.data(), grad_source_psiM.data(), psiMinv.size());
+    // psiMinv.cols() can be different from NumOrbitals
```

Comment:
This comment is a bit confusing.  Isn't the issue that psiMinv.cols() could be padded out because we want lda to be an aligned size?  My understanding is that this is what all the transposing in the cuBLAS and lapack is really about, we're taking a column major matrix with column padding but lapack interprets that as the transpose, i.e. that memory and padding is to it row major and the padding is on the "rows"

maybe
```
// psiMinv padding for SIMD row alignment is explicit i.e.
// psiMinv.cols() >= psiMinv.rows()  &&  psiMinv.cols() * sizeof(T) == is a multiple of QMC_SIMD_ALIGNMENT
// The actual Matrix with n = num_orbitals is stored in the square sublock psiMinv.rows() x psiMinv.rows().

Reply:
psiMinv documentation added.